### PR TITLE
[DRAFT] 2930 Add Half support to Microsoft.Data.Analysis

### LINKED ISSUE #2930: Add Half support to Microsoft.Data.Analysis
With https://github.com/dotnet/runtime/issues/936 and https://github.com/dotnet/corefxlab/pull/2929, .NET is getting a `System.Half` type. We should support this type in `DataFrame`, just like all the other built-in value types.

cc @pgovind

### PR DESCRIPTION
**Pull request by kevin-wittmer**
Opening this as a draft PR to get initial feedback. Some of the unit tests are currently failing because the `Microsoft.ML.Data` library does not yet have support for the `Half` data type.

Fixes #2930

### CODE REVIEW

File: src/Microsoft.Data.Analysis/Converters.cs [hunk 1]
```diff
@@ -677,6 +796,13 @@ public float GetSingle(ulong value)
             return (float)value;
         }
     }
+    internal class HalfSingleConverter : ISingleConverter<Half>
+    {
+        public float GetSingle(Half value)
+        {
+            return (float)(float)value;
```
Comment (danmosemsft) [inline src/Microsoft.Data.Analysis/Converters.cs:803]: redundant cast?

Comment (danmosemsft) [reply src/Microsoft.Data.Analysis/Converters.cs:803]: oh, it's generated code. fine.

**Comment by kevin-wittmer** [comment]
@pgovind Can you review this PR please?

**Comment by pgovind** [comment]
Apologies, I completely missed the notification for this PR. Will review first thing tomorrow!

**Comment by kevin-wittmer** [comment]
@pgovind No worries, thank you!

**Comment by pgovind** [comment]
I'm halfway through reviewing this PR. Just a note to myself to move the PR to dotnet/machinelearning.

**Pull request closed without merging**